registry.go

```diff
@@ -32,6 +34,59 @@ func (r *Registry) Add(t *Task, opts ...TaskOption) *Task {
 		panic(fmt.Sprintf("Duplicate task registered: %s", t.Name))
 	}
 	r.definitions[t.Name] = t
+
+	r.flagsByTask[t.Name] = map[string]TaskFlag{}
+	for _, flag := range t.Flags {
+		// Validate that the following task TaskFlag fields are provided:
+		// - LongName or ShortName
+		// - Disallow --help/-h flags
+		// - Description
+		// - ValueType
+		if flag.LongName == "" && flag.ShortName == 0 {
+			panic(fmt.Sprintf("Neither LongName nor ShortName are defined for an flag on `%s`. At least one name must be defined", t.Name))
+		}
+
+		if flag.LongName != "" && flag.LongName == "help" {
+			panic("The long flag name `help` is reserved")
+		}
+
+		if flag.ShortName != 0 && string(flag.ShortName) == "h" {
+			panic("The short flag name `h` is reserved")
+		}
+
+		if flag.Description == "" {
+			var optionName string
+			if flag.LongName != "" {
+				optionName = flag.LongName
+			} else {
+				optionName = string(flag.ShortName)
+			}
+			panic(fmt.Sprintf("Please provide a description for the `%s` flag", optionName))
+		}
+
+		if flag.ValueType != StringTypeFlag && flag.ValueType != BoolTypeFlag && flag.ValueType != IntTypeFlag {
+			panic("Please set the flag ValueType with `taskrunner.StringTypeFlag`, `taskrunner.BoolTypeFlag` or `taskrunner.IntTypeFlag`.")
+		}
+
+		// Validate that there are no duplicate LongNames
```

Comment:
nit - complete sentences so end with a period :)

Reply:
What is this, grammar school??? Jk jk